tools/editor/plugins/canvas_item_editor_plugin.h

```diff
@@ -35,13 +35,19 @@
 #include "scene/gui/panel_container.h"
 #include "scene/gui/box_container.h"
 #include "scene/2d/canvas_item.h"
-
+#include "scene/2d/light_2d.h"
```

Comment:
it is not bad practice.
it can have some advantages like you know all the dependencies of your code just by looking at it's header.

and anyway, most of the time, the class you declare in your new header has some dependencies that require you to include other headers.

here is interesting article about this topic : http://www.cplusplus.com/forum/articles/10627/